metrics/web/docs/analysis/deep_links/01_deep_links_analysis.md

```diff
@@ -0,0 +1,221 @@
+
+Now, let's review the listed approaches.
+
+### Deep Links Integration Using ChangeNotifier
+The main idea of this approach is to introduce a new `ChangeNotifier`, let's call it a `DeepLinksNotifier`. 
+
+__Note: the naming is a work in progress and may be changed during the `Design` phase.__
+
+The `DeepLinksNotifier` is responsible for:
+- Processing raw deep links provided by the `NavigationNotifier` into page-specific deep links;
+- Providing page-specific deep links to the corresponding page `ChangeNotifier`s to handle them;
+- Handling page-specific deep links updates provided by page `ChangeNotifier`s;
+- Providing deep link updates to the `NavigationNotifier`, which updates the browser history state to save a deep link update in a URL.
+
+#### Pros
+- Well segregated responsibility between components;
+- Extensible approach - meaning that we can easily add new page-specific deep links or extend them;
+- Easy to test, because of the responsibility segregation.
+
+#### Cons
+- Requires implementing new `ChangeNotifier`s and extending existing ones.
+
+### Deep Links Integration Using Route Parameters
+The main idea of this approach is to dispatch deep links using page parameters, which are then handled by the pages. For example, if a `DashboardPage` includes a project group selection menu, we can add a `projectGroupName` parameter to a `DashboardPage`, which is then handled by the page itself.
+
+However, according to the ["UI elements"](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#ui-elements) section of the ["Metrics Web Presentation Layer"](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md) document, a `Page` is a widget, whose responsibility is a proper Metrics widgets combining, thus, handling any parameters is out of scope of the `Page`'s responsibilities.
+
+#### Pros
+- Does not require implementing new or extending existing `ChangeNotifier`s.
+
+#### Cons
+- Does not satisfy the existing architecture requirements;
+- Harder to test;
+- Adding parameters handling may increase the `page` widgets size, and decrease the code readability.
+
+### Conclusion
+According to the comparison above, we choose the [Deep Links Integration Using ChangeNotifier](#deep-links-integration-using-changenotifier) approach, as it satisfies the architectural requirements, simplifies testing, and highly extensible.
+
+### Prototyping
+> Create a simple prototype to confirm that implementing this feature is possible.
+
+#### Processing raw deep links provided by the `NavigationNotifier` into page-specific deep links
+```dart
```

Comment:
Would it be great to introduce each code snippet with a short sentence describing the contents of this code snippet?